### PR TITLE
Coverity 1524619: unexpected control flow (alternate fix)

### PR DESCRIPTION
**Pull request by mattcaswell**
Using "continue" in a do...while(0) loop is pointless. The original intent was that the loop would continue to a second iteration in this case.

This is an alternative fix to #20669

### CODE REVIEW

File: apps/s_client.c [hunk 1]
```diff
@@ -3827,7 +3827,7 @@ static int user_data_process(struct user_data_st *user_data, size_t *len,
         char *cmd_start = buf_start;
 
         cmd_start[outlen] = '\0';
-        do {
+        for (;;) {
             cmd_start = strstr(cmd_start, "{");
```
Comment (t8m) [inline apps/s_client.c:3831]: When you're at it, you could also replace these single character strstr() calls with strchr() calls.

**Review by paulidale — APPROVED** [review]
With or without the strstr fix (better with though).

**Comment by mattcaswell** [comment]
strstr->strchr fix applied.

@t8m please re-review.

**Review by t8m — APPROVED** [review]

**Comment by mattcaswell** [comment]
Setting "approval done" on this, since the only change since @paulidale's approval was the strstr->strchr fix. @paulidale stated his approval stood regardless of whether or not that change was made.

**Pull request closed without merging**